src/core/execute.c

```diff
@@ -684,6 +684,25 @@ _printf_(2, 3) static int write_confirm_message(const char *vc, const char *form
         return 0;
 }
 
+static int write_confirm_error(int err, const char *vc) {
+        const char *tail = "assuming positive response.\n";
+        _cleanup_close_ int fd = -1;
+
+        assert(vc);
+        assert(err < 0);
+
+        fd = open_terminal(vc, O_WRONLY|O_NOCTTY|O_CLOEXEC);
+        if (fd < 0)
+                return fd;
+
+        if (err == -ETIMEDOUT)
+                dprintf(fd, "Confirmation question timed out, %s", tail);
+        else
+                dprintf(fd, "Couldn't ask confirmation: %s, %s", strerror(-err), tail);
```

Comment:
please do not introduce new uses of strerror()... We try to get rid of it entirely in the codebase, due the general incompatibility with threads. Please use %m or strerror_r(). In this case %m sounds like the better choice, just explicitly set `errno = -err` first.